File: 2023-2_9335-finteam5_python/BoggleClient.py
import sys
from omniORB import CORBA
import CosNaming
import BoggleApp
from GameClient import GameClient

def main():
    # Initialize the ORB
    orb = CORBA.ORB_init(sys.argv, CORBA.ORB_ID)

    # Obtain a reference to the root naming context
    try:
        objRef = orb.resolve_initial_references("NameService")
        rootContext = objRef._narrow(CosNaming.NamingContext)
        if rootContext is None:
            raise Exception("Failed to find the root naming context")
    except Exception as e:
        print(e)
        sys.exit(1)

    # Resolve the object reference in the naming context
    name = [CosNaming.NameComponent("Players", "")]
    try:
        objRef = rootContext.resolve(name)
    except CosNaming.NamingContext.NotFound:
        print("No Players found")
        sys.exit(1)

    # Narrow the object to the Boggle PlayerServant
    stub = objRef._narrow(BoggleApp.PlayerServant)

    # Create an instance of GameClient and pass the stub
    game_client = GameClient(stub)
    game_client.run()

if __name__ == "__main__":
    main()

File: 2023-2_9335-finteam5_java/src/main/java/Server_Client/BoggleClient.java
package Server_Client;

import BoggleApp.PlayerServant;
import BoggleApp.PlayerServantHelper;
import Runner.LoginControllerRunner;
import javafx.stage.Stage;
import org.omg.CORBA.ORB;
import org.omg.CORBA.ORBPackage.InvalidName;
import org.omg.CosNaming.NamingContextExt;
import org.omg.CosNaming.NamingContextExtHelper;
import org.omg.CosNaming.NamingContextPackage.CannotProceed;
import org.omg.CosNaming.NamingContextPackage.NotFound;

public class BoggleClient
{
    public static PlayerServant stub;
    public static void main(String[] args) {
        try {
            // create and initialize the ORB
            String[] args2 = {"-ORBInitialPort", "5151", "-ORBInitialHost", "10.135.135.194"};

            ORB orb = ORB.init(args2, null);
            org.omg.CORBA.Object objRef = orb.resolve_initial_references("NameService");
            NamingContextExt ncRef = NamingContextExtHelper.narrow(objRef);

            stub = PlayerServantHelper.narrow(ncRef.resolve_str("Players"));
            LoginControllerRunner.setStub(stub);

            new Thread(() -> LoginControllerRunner.launch(LoginControllerRunner.class)).start();
            System.out.println("Running Client");

        } catch (InvalidName | NotFound | org.omg.CosNaming.NamingContextPackage.InvalidName | CannotProceed invalidName) {
            invalidName.printStackTrace();
        } catch (Exception e) {
            throw new RuntimeException(e);
        }

    }
}
